airgun/browser.py

```diff
@@ -409,3 +412,49 @@ def save_downloaded_file(self, file_uri=None, save_path=None):
         self.url = current_url
         self.plugin.ensure_page_safe()
         return file_path
+
+    def check_alert(self, locator):
+        try:
+            return self.move_to_element(locator).is_displayed()
+        except NoSuchElementException:
+            self.logger.info(f'No {locator} type alert detected !')
+        return False
+
+    def get_alert(self):
+        """Returns the current alert/PF4 alert object.
+
+        Raises:
+            :py:class:`selenium.common.exceptions.NoAlertPresentException`
+        """
+        pf4_modal_locator = "//div[@data-ouia-component-type='PF4/ModalContent']"
+        modal_locator = "//div[@class='modal-content']"
+        modal_map = {pf4_modal_locator: Pf4ConfirmationDialog, modal_locator: ConfirmationDialog}
+        if not self.handles_alerts:
+            return None
+        try:
+            return self.selenium.switch_to.alert
+        except NoAlertPresentException:
```

Comment:
Great pattern here instead of trying is_displayed first.

Reply:
there is not displayed method for checking alerts, these only exists in the web element